src/include/uthenticode.h

```diff
@@ -156,8 +156,18 @@ class Certificate {
   /* TODO: Maybe add data_ and get_data(), with data_ populated from i2d_X509.
    */
 
+#ifndef UTHENTICODE_DEFAULT_XN_FLAGS
+  static constexpr unsigned long const default_xn_flags =
+      XN_FLAG_RFC2253 | ASN1_STRFLGS_UTF8_CONVERT;
+#else
+  static constexpr unsigned long const default_xn_flags = (UTHENTICODE_DEFAULT_XN_FLAGS);
+#endif
+  static_assert((default_xn_flags & XN_FLAG_COMPAT) == 0,
+                "Logic is incompatible with XN_FLAG_COMPAT");
+
  private:
-  Certificate(X509 *cert);
+  explicit Certificate(X509 *cert);
+  Certificate(X509 *cert, unsigned long xn_flags);
```

Comment:
I'd rather not expose a custom ctor here -- it effectively binds us to the OpenSSL ABI (via `xn_flags`, plus formalizing that we support OpenSSL's weird stringification, which wasn't intentional), and exposes a decision point that I suspect 99% of people using this library don't need.

If the goal is to change the default to `XN_FLAG_RFC2253` + UTF-8 conversion, let's just do that and then I'll perform a major release to indicate potential breakage.

Reply:
Amended the original commit. Please have a look if this resolves the issue for you, @woodruffw.